### PR TITLE
Import trace replay into lib context

### PR DESCRIPTION
**Pull request by maiste**
- Migrate `manage_actions.exe`
- Import `trace_replay` et `raw_actions_to_replayable`
- Check into `to_replayable`:
  - Inject an `init` if it is missing at the beginning of the slice
  - There is only __ONE__ `init`
  - `Commit_genesis` is the last operation on the first block if level = 0
- `Commit_genesis` is split into `Commit_genesis_start` and `Commit_genesis_end` to prevent error with `patch_context` in the replay
- Improving the pretty-printer when there is a message in the header:
  - If it can be parsed as a JSON, it introduces the column with key | value
  - If all the JSON have the same entries,  it replaces the value with `<same>` nin table0
  - Display a Warning if it is not a JSON format on stderr
  - In the setup summary, it displays `<same>` if columns have the same value
- Magics have been replaced and the code is referenced as _V0_
- Check that summary.json is writable otherwise it fails at the beginning of the replay
- It uses the Tezos default node option for the __GC__
- Raw traces instantiate writers in the `init` function
- Stats traces support the path to match the ones from migration
- Correct the problem of duration = 0
- Print a warning on stderr if `TEZOS_CONTEXT` env value is empty or wrong
- Remove `cb` from `manage_stats`
- ~~Rename `input*output` to `neg*pos` (see [here](https://www.schoolofhaskell.com/user/commercial/content/covariance-contravariance))~~ (Doesn't make that much sens)
- Add documentation about how to run traces
- Support `restore_context` and `dump_context`
- Update headers with the right values
- Support `Optint63` with `repr.0.6.0` to reduce trace sizes

### CODE REVIEW

File: src/lib_context/recording/trace_stats_summary_pp.ml [hunk 1]
```diff
@@ -114,138 +114,17 @@ let pp_scalar_fixed ppf (format, v) =
...
+  end)
 
-module Table0 = struct
-  let summary_config_entries =
+  let no_value = "\xC3\xB8"
+
+  let static_summary_config_entries =
     [
-      `Message;
-      `Hostname;
```
Comment (Ngoguey42) [inline src/lib_context/recording/trace_stats_summary_pp.ml]: Missing Hostname

File: src/lib_context/recording/trace_stats_summary_pp.ml [hunk 2]
```diff
@@ -377,6 +296,12 @@ module Table0 = struct
     let separator_row =
       [List.init (summary_name_length + 1) (Fun.const "--")]
     in
+    let dyn_summary_config_entries : [> `Key of string] list =
+      extract_all_keys summaries
+    in
```
Comment (Ngoguey42) [inline src/lib_context/recording/trace_stats_summary_pp.ml]: I think that `` `Message`` should be added to `dyn_summary_config_entries` if at least 1 summary is not parsable

File: src/lib_context/recording/trace_stats_summary_pp.ml [hunk 3]
```diff
@@ -343,6 +240,28 @@ module Table0 = struct
...
+    | Some json -> (
+        match parse_json json with
+        | Ok kv ->
+            List.fold_left
+              (fun acc (json_key, _) -> MsgSet.add json_key acc)
+              set
+              kv
+        | _ -> set)
+
+  let extract_all_keys summaries : [> `Key of string] list =
```
Comment (Ngoguey42) [inline src/lib_context/recording/trace_stats_summary_pp.ml]: Putting the type here highlight your typing wish, which is a good thing by itself. However the compiler will only use it to extend to lower bound. If there's an upper bound it won't remove it

Comment (Ngoguey42) [reply src/lib_context/recording/trace_stats_summary_pp.ml]:
```
let a : [> `B ] list  = [ ` A ]
```
gives

```
val a : [> `A | `B ] list
```

**Comment by maiste** [comment]
This PR can be closed as everything lives in `new-action-trace-recording` now.

**Pull request closed without merging**